src/components/Datatable/Table/Table.tsx

```diff
@@ -49,15 +49,15 @@ const StyledTableBody = styled.thead<{ isTableSticky: boolean }>`
     `};
 `;
 
-const Table: React.FC<TableProps<Data>> = ({
+const Table = <D extends Record<string, unknown>>({
```

Comment:
Please, what is the advantage of using Record type here? If I will add new prop then it still fails, right?

Reply:
`D` is generic type and we need to add some constraints to it because it would be too wide for typescript. This way we tell TS that the D will have some shape of JS object where keys will be strings and we don't care about values type. 
This will give us a way how to pass type which describes the table data type which will be different for each table.
We can use it then like:
```js
type Data = {
  id: number;
  ipAddress: string;
  ...
}

...

return <Table<Data> ... />
```
And the `Table` component will know type will have the data we pass in. 

The `Record<string, unknown>` notation we use instead of `{}` because `{}` means any non-nullish value but Record<> means any object.

Reply:
Thanks for clear explanation! 🙏